book/src/simple_application/pong_system.md

```diff
@@ -0,0 +1,112 @@
+# The Game System
+
+In this stage we are going to add a system to our world that will run the main
+game mechanics. We won't implement everything, but we will see how the system
+reads the state and updates it.
+
+First thing we want to do is create some components that contain our game data,
+rather than render/input. So far we've use built-in components, this is the
+first time we write them ourselves.
+
+```rust
+struct Ball {
+    pub position: [f32; 2],
+    pub velocity: [f32; 2],
+    pub size: f32,
+}
+
+impl Ball {
+    pub fn new() -> Ball {
+        Ball {
+            position: [0.0, 0.0],
+            velocity: [-1.0, -1.0],
+            size: 1.0,
+        }
+    }
+}
+
+impl Component for Ball {
+    type Storage = VecStorage<Ball>;
+}
+```
+
+Let's go through what we've done here. Firstly, we create a datatype for our
+ball. It contains 2D position and velocity information, and a size, and has a
+simple constructor.
+
+We then implement `Component` for our data structure, allowing it to be used as
+a component in our *ecs* world. To do this, we need to say how we store the
+data. `VecStorage` is probably the simplest way, and according to the
+documentation optimal for dense data (components present on most entities).
+Since we're only going to have 3 entities (2 paddles and a ball) it probably
+doesn't matter about performance.
```

Comment:
Maybe:
~~Since we're only going to have 3 entities (2 paddles and a ball) it probably
doesn't matter about performance.~~
Since we're only going to have 3 entities (2 paddles and a ball) we don't have to worry about performance that much.